plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c

```diff
@@ -449,9 +450,19 @@ fu_logitech_bulkcontroller_device_write_firmware(FuDevice *device,
 	if (fw == NULL)
 		return FALSE;
 
-	/* Sending INIT */
-	if (!fu_logitech_bulkcontroller_device_send_upd_cmd(self, CMD_INIT, NULL, error)) {
-		g_prefix_error(error, "error in writing init transfer packet: ");
+	/* Sending INIT. Retry if device is not in IDLE state to receive the file */
+	do {
+		if (!fu_logitech_bulkcontroller_device_send_upd_cmd(self, CMD_INIT, NULL, error)) {
+			g_prefix_error(error,
+				       "error in writing init transfer packet: retrying... ");
+			continue;
+		}
+		break;
+	} while (--init_retry);
```

Comment:
There are multiple problems with this, including re-using the GError and not setting the error on failure. Could you use `fu_device_retry()` instead? The idea with that is you create a callback (perhaps `fu_logitech_bulkcontroller_device_send_upd_cmd_cb`) that does the action, and then `fu_device_retry()` handles all the things like waiting for a set amount of time (say 20ms) and saving the GError in the appropriate way.

Reply:
Fixed